libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -57,7 +57,7 @@ public MapboxTelemetry(Context context, String accessToken, String userAgent) {
     this.userAgent = userAgent;
     AlarmReceiver alarmReceiver = obtainAlarmReceiver();
     this.schedulerFlusher = new SchedulerFlusherFactory(applicationContext, alarmReceiver).supply();
-    this.telemetryEnabler = new TelemetryEnabler(true);
+    this.telemetryEnabler = new TelemetryEnabler(true, applicationContext);
```

Comment:
i'd rather suggest to deprecate this class going forward. it's extremely fragile and not thread safe.